docs/reference/github/workflow/branches.md

```diff
@@ -0,0 +1,148 @@
+# Developing on Branches
+
+We use branching to diverge from the main line of development and work on issues without directly
+modifying the main line. This ensures that our codebase is robust and avoids merge conflicts
+when multiple people are developing at the same time. For a quick rundown on branching in git,
+consult the official [git documentation](https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell).
+
+## Creating a branch
+
+There are three possible methods to create a branch for your next issue:
+
+!!! caution
+
+    When creating branches locally, it uses your **local copy** to create the new branch. Remember to do a `git pull` 
+    if you intend on using the latest changes from the remote branch you are creating from.
+
+=== "From Current Branch"
+
+    This method creates a branch from the current branch that you are on and checks out
+    to that branch automatically.
+
+    ``` bash
+    # Create new branch from current branch and checkout to the new branch
+    git checkout -b <branch_name>
+    ```
+
+=== "From Specific Branch"
+
+    This method creates a branch from a specified branch.
+
+    ``` bash
+    # Create the branch
+    git branch <branch_name> <source_branch_name>
+
+    # Checkout to the new branch
+    git checkout <branch_name>
+    ```
+
+=== "From GitHub GUI"
+
+    ![image](../../../assets/images/github/workflow/branch_from_gui.png)
+    ![image](../../../assets/images/github/workflow/name_branch_gui.png)
+    
+    Under the `Development` section on the issue page, click `Create a branch`. Afterwards, you will see
+    a prompt asking you to name your branch. 
+
+    Once you created your branch, it will be available on the remote repository. To gain access to it
+    on your local machine, do a `git pull` and checkout to that branch.
+
+## Branch naming convention
```

Comment:
I think that we should enforce this because this allows us to know who is working on what issue in a specific branch. There have been far too many instances where I have seen mysterious branches and I don't even know what they were used for. It is one extra thing to remember, but I believe that it's good practice and will serve us well.

Reply:
I agree, but might I suggest using one's GitHub username instead of first name.

Also, some branches might not be tied to issues so I don't think there has to be any requirements for after the `/`.
What I like to do is keep the branch name short and have a better description in the PR title

Reply:
Ok, I will change it to GitHub username and make a note on the issue number being optional.